packages/web3/src/utils/exchange.ts

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressType, DUST_AMOUNT, addressFromPublicKey, addressFromScript, binToHex, bs58, hexToBinUnsafe } from '..'
+import { Transaction } from '../api/api-alephium'
+import { Address } from '../signer'
+
+enum UnlockScriptType {
+  P2PKH = 0x00,
+  P2MPKH = 0x01,
+  P2SH = 0x02
+}
+
+export function getAddressFromUnlockScript(unlockScript: string): Address {
+  const decoded = hexToBinUnsafe(unlockScript)
+  if (decoded.length === 0) throw new Error('UnlockScript is empty')
+  const unlockScriptType = decoded[0]
+  const unlockScriptBody = decoded.slice(1)
+
+  if (unlockScriptType === UnlockScriptType.P2PKH) {
+    return addressFromPublicKey(binToHex(unlockScriptBody))
+  } else if (unlockScriptType === UnlockScriptType.P2MPKH) {
+    throw new Error('Multisig address is not supported')
+  } else if (unlockScriptType === UnlockScriptType.P2SH) {
+    // FIXEME: for now we assume that the params is empty, so we need to
```

Comment:
Because the unlock script consists of both script and params, we need to use the codec in this PR: https://github.com/alephium/alephium-web3/pull/276. For now we only have the schnorr script, and its params are empty.